### ISSUE TITLE
Adventure map, bad positioning of spawned monsters after "week of"

### ISSUE THREAD

**Issue by Branikolog** [ihhub/fheroes2#4426]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

![image](https://user-images.githubusercontent.com/55348946/137456985-abd444fc-6d24-4002-b03c-9f79a87736ad.png)
Related closed issue: #2363

### Save file

Campaign save file:
[bug_Golem-spawn-Roland.zip](https://github.com/ihhub/fheroes2/files/7352071/bug_Golem-spawn-Roland.zip)

**Comment by ihhub** [comment]
Hi @Branikolog , can you please check this issue as we fixed many passability issues in the past?

**Comment by Branikolog** [comment]
Hi, @ihhub !
Okay! I'll try to do my best.
By the way, I'm concerned about these weeks shedule. While investigating recently close issue with town growth, I've noticed, that each time I've started test map, first week was a "week of .... +5", and new month 100% started with double growth of particular creature... I suppose these events should take more random place in "heroes time", shouldn't they?.
